src/WebJobs.Script/Workers/Http/HttpFunctionInvocationDispatcher.cs

```diff
@@ -185,5 +185,10 @@ public Task ShutdownAsync()
         {
             return Task.CompletedTask;
         }
+
+        public async Task RestartAsync()
+        {
```

Comment:
Add `State = FunctionInvocationDispatcherState.WorkerProcessRestarting;` here too

Reply:
State is already being changed as the first step within DisposeAndRestartWorkerChannel() which is why I didn't add it here... This was not the case for RpcFunctionInvocationDispatcher though.

Reply:
Ahh got it! I think it's in a different place for rpc because we only want to delay invocations if there are 0 workers alive. Sounds good, thanks!